resources/assets/js/views/Preview.vue

```diff
@@ -286,7 +285,15 @@ export default {
 
 				// the defined region was not loaded in our region definitions
 				else {
-					$this.layoutErrors.push(`The defined region '${regionName}' was not found in our loaded region definitions`);
+					this.layoutErrors.push(`The defined region '${regionDefinitionName}' was not found in our loaded region definitions`);
+				}
+			});
+
+			// another loop through the data regions to alert the user if there are regions that are not defined
+			Object.keys(this.pageData.blocks).forEach((regionDataName) => {
```

Comment:
For completeness we should also check that there aren't any additional sections in any of the regions.

e.g. if a region can have 2 sections, and the pagedata has these sections but also has an additional section then this is not caught anywhere.

At a minimum in the section checking above maybe check that there aren't more sections than allowed in each region.

Reply:
this has been added in